lib/internal/modules/esm/module_job.js

```diff
@@ -92,7 +94,22 @@ class ModuleJob {
         this.module.instantiate();
       }
     } catch (e) {
-      decorateErrorStack(e);
+      if (e.message.includes('does not provide an export named')) {
+        const rootURL = this.module.url;
+        const childSpecifier = StringPrototypeMatch(
+          e.message,
+          // Match the first string in either single or double quotes
```

Comment:
this is still not guaranteed to match; the if check doesn’t ensure there’s quotes. If it’s null, this would crash on the [1].

Reply:
We can check to ensure but I'm 99% sure this error cannot be reached except when the specifier fails. Will write a bunch of tests though to ensure my assumptions and update accordingly

Reply:
fair enough; it still seems safer to me to use error codes to pivot, and to store this information as a property on the error rather than parsing it out of the error message